packages/expo-modules-core/src/web/index.web.ts

```diff
@@ -1 +1 @@
-import './CoreModule';
+export * from './CoreModule';
```

Comment:
Now I'm wondering why does changing import to export change the behavior?
Isn't it that we just need to import `web/index` before `NativeModule` and `SharedObject` are imported?

Reply:
>Isn't it that we just need to import web/index before NativeModule and SharedObject are imported?
this change doesn't actually work. 

The core reason is due to `globalThis`, which actually differs between Node.js and the browser.

On the Next.js server side (Node.js), it refers to `global`.
In the browser environment, it refers to `window`.

During SSR, the code may be executed multiple times, and the state of `globalThis` may not be persisted.

For example:
If I change the CodeModule.js code to the code below, it will work because the `globalThis` state can be persisted.
```
if (typeof window !== "undefined") {
    window.expo = {      
        EventEmitter,
        ...
    }
  } else if (typeof global !== "undefined") {
    global.expo = {
        EventEmitter,
        ...
    }
}

```

Reply:
I'm now simply changing the import order to fix it: 

When I execute `import './web/index.js'` on the server side, the module `./web/index.js` is loaded first. Since `./web/index.js` directly exports all content from `./CodeModules.js`, this means that `./core/index.js` will be executed and will correctly set `globalThis.expo` to `{ NativeModule, ....}`.

Reply:
lmk if you have any questions!

Reply:
Gotcha, though I still wonder if moving this import 👇 wouldn't be sufficient to solve the issue? https://github.com/expo/expo/blob/cdae328d76391ba3a93fc169f6d7fa840a1a108c/packages/expo-modules-core/src/index.ts#L14

Reply:
Unfortunately, it doesn't work.